src/VASSAL/counters/Stack.java

```diff
@@ -76,14 +77,25 @@ public Stack(GamePiece p) {
    * @return an Enumeration of the pieces in the stack, from the bottom up This
    *         is a clone of the contents so add/remove operations during read
    *         won't affect it.
-   * @deprecated
+   * @deprecated use {@link #getPiecesAsList()}
    */
   @Deprecated
   public Enumeration<GamePiece> getPieces() {
     return new EnumeratedIterator<>(new AllPieceIterator());
     //    return new AllPieceEnum();
   }
 
+  /**
+   * @return a defensive copy of the {@link List} of {@link GamePiece}s contained in this {@link Stack}
+   */
```

Comment:
I need to think about whether a copy is the right thing, or if what we want is (instead, or additionally) a way to get an unmodifiable view.

Reply:
I can't find any place that we use `Stack.getPiecesIterator()` just to fill a list of pieces. Returning an unmodifiable view of `contents` is what we want. If something which doesn't exist now really needs its own copy of the list, it can make one itself.